server/server.go

```diff
@@ -0,0 +1,611 @@
+package server
```

Comment:
@Dotunj, I think we want to make this package re-usable as well. So that other services can use them too.

Like moving this guy into `internal/pkg/server`

Reply:
Secondly, we need to decouple the server from the router/handler. This is needed so that other services can bring their handlers. 

An API like:

```Golang
 
   apiHandler := buildRoutes()

   server := &Server{
      Handler: apiHandler,
   }
```
In the client code.